ClinicalTrial_SE_curation.Rmd

```diff
@@ -0,0 +1,320 @@
+library(GEOquery)
+library(SummarizedExperiment)
+library(qs)
+library(affy)
+```
+
+```{r}
+geo_rma<- function (raw_tar, geo_path , brain_path){
+  
+  # Download CEL files
+  cel.file <- paste(work_dir, raw_tar, sep="/")
+  options(download.file.method="curl", download.file.extra="-k -L")
+  download.file(geo_path, cel.file)
+  
+  # Unpack the CEL files
+  exit_dir <- sub("\\..*", "", raw_tar)
+  untar(raw_tar, exdir=exit_dir)
+  cel.files<-list.files(paste(exit_dir, "/", sep=""), pattern = "gz")
+  sapply(paste(exit_dir, cel.files, sep="/"), GEOquery::gunzip)
+  
+  # Install the cdf brainarray package
+  pack_name <- tail(str_split(brain_path, "/")[[1]], n=1) 
+  cdf.file <- paste(work_dir, pack_name, sep="/")
+  options(download.file.method="curl", download.file.extra="-k -L")
+  download.file( brain_path, cdf.file)
+  install.packages(pack_name, repos = NULL, type = "source")
+  
+  # RMA normalization
+  cdf <- sub("_.*","",pack_name)
+  cels <- affy::list.celfiles(paste(exit_dir, "/", sep=""), full.names = TRUE)
+  rma.norm <- affy::justRMA(filenames = cels, verbose = TRUE, cdfname = cdf)
+  
+  return(rma.norm)
+  
+} 
+```
+
+```{r}
+# ExpressionSet 
+ESet <-  function(rma.norm, GSE , delim, GPL, delim2=NA, cancer_type=NA, treatment = NA, response = NA, survival_type = NA,
+                  survival_time = NA,  survival_time_unit=NA, event_occured = NA){
+  
+  # Assay data
+  assay_data <- as.data.frame(exprs(rma.norm)) 
+  rownames(assay_data) <-sub("_at","",rownames(assay_data)) 
+  colnames(assay_data) <-sub(delim ,"",colnames(assay_data)) 
+  if (!identical(c(grep("AFFX", rownames(assay_data))) , integer(0))) { 
+    assay_data <- assay_data[-c(grep("AFFX", rownames(assay_data))),] } # Removing control genes (which start with "AFFX")
+  
+  # Pheno data
+  gset <- GEOquery::getGEO(GSE, GSEMatrix =TRUE, getGPL=FALSE)
+  if (length(gset) > 1) idx <- grep(GPL, attr(gset, "names")) else idx <- 1
+  gset <- gset[[idx]]
+  
+  if (all(colnames(assay_data) %in% rownames(gset@phenoData@data))){
+    pheno_data <- gset@phenoData@data[colnames(assay_data),]
+  } else {colnames(assay_data) <-sub(delim2 ,"",colnames(assay_data))
+  pheno_data <- gset@phenoData@data[colnames(assay_data),]}
+  
+  # adding standard columns:
+  pheno_data$unique_patient_ID <- rownames(pheno_data)
+  prior <- c("cancer_type","treatment", "response", "survival_type", "survival_time", "survival_time_unit", "event_occured")
+  
+  for (c in prior){
+    if (is.na(get(c))){
+      pheno_data[[c]] <- NA
+    } else {
+      pheno_data[[c]] <- pheno_data [,  get(c)] 
+    }
+  }
+  
+  # Ordering columns based on priority
+  pheno_data <- pheno_data[, c("unique_patient_ID", prior, colnames(pheno_data)[!colnames(pheno_data) %in% prior])]
+  
+  # Replacing "NA" with NA_character_ entries
+  pheno_data[pheno_data=="NA"]=NA
+  
+  
+  # Feature data
+  feat_data <- merge(features_gene, data.frame(gene_id = rownames(assay_data)), all.y=T, by = "gene_id")
```

Comment:
Best Practices:

Never use `T` or `F` in an R script or function, they are only for interactive use. Take the time to write out `TRUE` or `FALSE` to enhance the clarity of your code.

Reply:
Done